src/main/java/org/wildfly/naming/client/WildFlyRootContext.java

```diff
@@ -97,58 +100,90 @@ public NameParser getNativeNameParser() {
         return NAME_PARSER;
     }
 
-    protected Object lookupNative(final Name name) throws NamingException {
+    @Override
+    public Object lookup(String name) throws NamingException {
+        return lookup(new CompositeName(name));
+    }
+
+    @Override
+    public Object lookup(Name name) throws NamingException {
+        final ReparsedName reparsedName = reparse(name);
+        return getProviderContext(reparsedName.getUrlScheme()).lookup(reparsedName.getName());
+    }
+
+    ReparsedName reparse(final Name origName) throws InvalidNameException {
+        final Name name = (Name) origName.clone();
         if (name.isEmpty()) {
+            return new ReparsedName(null, name);
+        }
+        final String first = name.get(0);
+        final int idx = first.indexOf(':');
+        final String urlScheme;
+        if (idx != -1) {
+            urlScheme = first.substring(0, idx);
+            final String segment = first.substring(idx+1);
+
+            name.remove(0);
+            if(segment.length()>0 || (origName.size()>1 && origName.get(1).length()>0)){
+                name.add(0, segment);
+            }
+        } else {
+            urlScheme = null;
+        }
+        return new ReparsedName(urlScheme, name);
+    }
+
+    protected Object lookupNative(final Name name) throws NamingException {
+        final ReparsedName reparsedName = reparse(name);
+        if (reparsedName.isEmpty()) {
             return new WildFlyRootContext(getEnvironment().clone());
         }
-        final URLSchemeName urlSchemeName = URLSchemeName.fromName(name);
-        return getProviderContext(urlSchemeName).lookup(urlSchemeName);
+        return getProviderContext(reparsedName.getUrlScheme()).lookup(reparsedName.getName());
     }
 
     protected void bindNative(final Name name, final Object obj) throws NamingException {
```

Comment:
If lookup() needs to be overridden, then bind()/rebind()/etc. also need to be overridden.